pages/learn/develop/runtime.md

```diff
@@ -249,74 +249,79 @@ At the time of writing there is an inconsistency in the behaviour of `/tmp` dire
-```Dockerfile
-RUN echo "LABEL=mysdcard /mnt/storage ext4 rw,relatime,discard,data=ordered 0 2" >> /etc/fstab
-```
-Modify your settings as appropriate (device identification, mount endpoint, file system, mount options), and see more information about the possible settings at the [fstab man page](http://man7.org/linux/man-pages/man5/fstab.5.html).
+**balenaOS kernel support**
 
-Then in your start script you need to create the mount directory and mount the device:
-```Bash
-mkdir -p /mnt/storage && mount /mnt/storage
-```
+Before you start it's a good idea to check if the balenaOS kernel you are running was compiled with support for the filesystem you want to use. To do so, you can run this command on the **host** which will produce a list of supported filesystems: `cat /proc/filesystems`.
 
-**Using systemd**
+If your filesystem is not supported you can contact us through our [forums](https://forums.balena.io/) and we will be glad to help.
 
-Normally systemd mounts entries from `/etc/fstab` on startup automatically, but running within Docker, it will only mount entries that are not block devices, such as `tmpfs` entries. For non-block devices, adding entries `/etc/fstab` is sufficient, for example in your Dockerfile:
-```Dockerfile
-RUN echo "tmpfs  /cache  tmpfs  rw,size=200M,nosuid,nodev,noexec  0 0" >> /etc/fstab
+__Note:__ You can get a list of device names, labels and filesystem types by running `lsblk -f` (both on the host or container).
+
+**Preparing the container**
+
+In order to be able to detect external media dynamically you will need to run the container on privileged mode and enable `udevd` on it. This can be easily done if you are using [balena base images](https://www.balena.io/docs/reference/base-images/base-images/#working-with-dynamically-plugged-devices) by:
+- Adding `privileged: true` to your container's service definition on the `docker-compose.yml` file
+- Adding `ENV UDEV=on` to your container's `Dockerfile`
+
+This will ensure that the host propagates udev events into the container, enabling us to manipulate the device from within it.
+
+**Mounting**
+
+To mount an external drive you can use Linux's `mount` command. You can either mount a device by it's device name (i.e: `/dev/sda1`):
+
+```bash
+mount -t <fstype> -o rw <device-name> <mount-point>
 ```
 
-For block devices (SD cards, USB sticks), `/etc/fstab` entries would result in this error at runtime: `Running in a container, ignoring fstab device entry for ...`. Instead, you have to use [systemd .mount files](https://www.freedesktop.org/software/systemd/man/systemd.mount.html). Let's assume you want to mount an external SD card to `/mnt/storage`. Then you have to create a file with the name `mnt-storage.mount` and the content such as:
+Or mount it by it's device label (i.e: `my-usb-drive`):
```

Comment:
Is mounting by UUID also supported?  (although I guess that probably doesn't make sense in a fleet-of-devices context)

Reply:
It is supported, however as you suggest it usually does not make much sense. Nonetheless, I'm changing this section slightly to account for this and also bringing back a general note on how to device selection. Thanks!